src/kernel.ts

```diff
@@ -448,6 +459,22 @@ class Kernel implements IKernel {
         future.handleMsg(msg);
       }
     }
+    if (msg.channel === 'iopub') {
+      switch(msg.header.msg_type) {
+        case 'status':
+          this._updateStatus(msg.content.execution_state);
+          break;
```

Comment:
What about passing all `comm_` messages to a `handleCommMessage` on the `CommManager`, and letting it deal with it as it sees fit? That will remove the need for public-but-not-part-of-the-interface methods.